assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js

```diff
@@ -127,13 +141,22 @@ function Media3pPane(props) {
       currentPaddingLeft - scrollbarWidth + 'px';
   }, [scrollbarWidth, refContainer]);
 
-  const { media } = useMedia3pForProvider('unsplash', ({ state }) => ({
-    media: state.media,
-  }));
+  /**
+   * Effectively performs a search, triggered at most every 500ms.
+   */
+  const [changeSearchTermDebounced] = useDebouncedCallback(() => {
+    search({ searchTerm: searchTermValue });
+  }, 500);
 
-  const onSearch = useCallback(() => {
-    // TODO(#2391): Perform search.
-  }, []);
+  /**
+   * Handle search input changes. Triggers with every keystroke.
+   *
+   * @param {string} value the new search term.
+   */
+  const onSearch = (value) => {
+    setSearchTermValue(value);
+    changeSearchTermDebounced();
```

Comment:
Wasn't this part of another PR? 🤔

Reply:
It was for the upload tab. This PR doesn't yet unify those 2 things (it's repeated now).